buildroot-external/package/hassio/Config.in

```diff
@@ -18,4 +18,10 @@ config BR2_PACKAGE_HASSIO_MACHINE
 	help
 	  Machine to pull containers for (used for landing page).
 
+config BR2_PACKAGE_HASSIO_CHANNEL
+	string "Default Channel"
+	default "stable"
+	help
+	  Channel to use by default.
+
```

Comment:
We could use a `choice` like this:
```suggestion
choice
	prompt "Default Channel"
	default BR2_PACKAGE_HASSIO_CHANNEL_STABLE
	help
	  Channel to use by default.

config BR2_PACKAGE_HASSIO_CHANNEL_STABLE
	bool "Stable"
	help
	  Stable channel.

config BR2_PACKAGE_HASSIO_CHANNEL_BETA
	bool "Beta"
	help
	  Beta channel.

config BR2_PACKAGE_HASSIO_CHANNEL_DEV
	bool "Dev"
	help
	  Dev channel.

endchoice

config BR2_PACKAGE_HASSIO_CHANNEL
	string
	default "stable" if BR2_PACKAGE_HASSIO_CHANNEL_STABLE
	default "beta" if BR2_PACKAGE_HASSIO_CHANNEL_BETA
	default "dev" if BR2_PACKAGE_HASSIO_CHANNEL_DEV

```

Then to change the channel you can just append the desired value, e.g. `echo "BR2_PACKAGE_HASSIO_CHANNEL_DEV=y" >> .config` and the Kconfig will adjust pick the last value from the `.config` and set the string value accordingly.

Reply:
I've implemented a variant using choice, but convert it in the makefile to the actual filename. With this we have all parts of the URL string in the makefile instead of a mix between Config.in and makefile.